src/shared/cloudformation/activation.ts

```diff
@@ -12,6 +12,14 @@ import { CloudFormationTemplateRegistryManager } from './templateRegistryManager
 
 export const TEMPLATE_FILE_GLOB_PATTERN = '**/template.{yaml,yml}'
 
+/**
+ * Match any file path that contains a .aws-sam folder. The way this works is:
+ * match anything that starts  with a '/' or '\', then '.aws-sam', then either
+ * a '/' or '\' followed by any number of characters or end of a string (so it
+ * matches both /.aws-sam or /.aws-sam/<any number of characters>)
+ */
+export const TEMPLATE_FILE_BANNED_PATTERN = /.*[/\\]\.aws-sam([/\\].*|$)/
```

Comment:
"exclude" is more conventional (inverse of "include") in the context of glob/regex patterns.
```suggestion
export const TEMPLATE_FILE_EXCLUDE_PATTERN = /.*[/\\]\.aws-sam([/\\].*|$)/
```

Reply:
> > Code cleanup `addTemplatesToRegistry` was redundant, the base function already had an unused quiet option. Use it in the one place `addTemplatesToRegistry` was used
> 
> I've also wondered if `templateRegistry.ts` and `templateRegistryManager.ts` could be combined. Not clear to me why they are separate classes.

Same (there's even an existing comment in the test), but that is outside of the scope of this PR